src/components/TimeTracker.tsx

```diff
@@ -1,8 +1,180 @@
-import React, { useState, useEffect, useRef } from "react";
+import { useState } from "react";
 import { useSelector, useDispatch } from "react-redux";
+import { AddTimeEntry } from "./AddTimeEntry";
+import { 
+    updateStartTime, 
+    updateEndTime, 
+    updateTaskName, 
+    updateDuration,
+    resetState
+} from "../redux/clockifySlice";
+import { 
+    calculateEndDate, 
+    getFormattedTime, 
+    calculateDays, 
+    formatTime 
+} from "../utils/dateFunctions";
+import { 
+    calculateTimeDifference, 
+    convertToHoursAndMinutes, 
+    isDurationLimitExceeded, 
+    convertDurationToHrsMinsSecs, 
+    calculateEndTime,
+} from "../utils/hoursAndMinutes";
+import { groupEntriesByWeek } from "../utils/groupEntriesByWeek";
+import { RootState, AppDispatch } from "../redux/store";
+import { FocusEvent, KeyboardEvent, Type } from "../types/types";
+import { checkString } from "../utils/checkString";
+import { createTimeEntry } from "../redux/clockifyThunk";
+import { TimeEntries } from "./TimeEntries";
 
 export function TimeTracker(){
+    const { isModalOpen, currentTask, selectedProject, selectedClient, data} = useSelector((state: RootState) => state.clockify)
+    const {startTime, endTime, duration, taskName} = currentTask
+
+    const timeStart = new Date(startTime)
+    const timeEnd = new Date(endTime)
+    
+    const [showProjects, setShowProjects] = useState(false)
+    const [startDateTime, setStartDateTime] = useState(getFormattedTime(timeStart))
+    const [endDateTime, setEndDateTime] = useState(getFormattedTime(timeEnd))
+    const [totalDuration, setDuration] = useState(duration)
+
+    const dispatch = useDispatch<AppDispatch>()
+
+    const handleStartTimeBlur = (type: Type, e: FocusEvent): void => {
```

Comment:
Too much logic to understand if we need to make changes here. See if possible to refactor this. First understand the steps of algorithm and then see if that can be simplified or ordered in such a way that it's easier to grasp for any readers, especially a future you.

Reply:
https://github.com/PreethiKalyani99/Clockify-Typescript/commit/5191e28172351d61ca7314d8442318a6b37f5c0f

Refactored blur functions logic.